File: ga/glm4_result_2024_6/result/1968B_cpp_glm/48_1968B.cpp
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let mut tokens = input.trim().split_whitespace();
        let _x: i32 = tokens.next().unwrap().parse().unwrap();
        let _k: i32 = tokens.next().unwrap().parse().unwrap();
        let a: String = tokens.next().unwrap().to_string();
        let b: String = tokens.next().unwrap().to_string();

        let mut ans = 0;
        for (i, c) in b.chars().enumerate() {
            if ans < a.len() && a.chars().nth(ans) == Some(c) {
                ans += 1;
            } else {
                break;
            }
        }

        println!("{}", ans);
    }
}

File: ga/4omini_result_2024_7/result/1968B_cpp_mini/48_1968B.rs
use std::io;

fn main() {
    let stdin = io::stdin();
    let mut input = String::new();
    stdin.read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        let mut input = String::new();
        stdin.read_line(&mut input).unwrap();
        let mut iter = input.split_whitespace();
        let x: i32 = iter.next().unwrap().parse().unwrap();
        let k: i32 = iter.next().unwrap().parse().unwrap();

        let mut a = String::new();
        let mut b = String::new();
        
        stdin.read_line(&mut a).unwrap();
        stdin.read_line(&mut b).unwrap();

        let mut ans = 0;
        let a_chars: Vec<_> = a.chars().collect();
        
        for i in b.chars() {
            if ans < a_chars.len() && a_chars[ans] == i {
                ans += 1;
            }
        }
        println!("{}", ans);
    }
}